platforms/allwinner-d1/core/src/drivers/spim.rs

```diff
@@ -27,9 +28,9 @@ pub struct Spim1 {
 ///
 /// - The `SPI_DBI``s register block must not be concurrently written to.
 /// - This function should be called only while running on an Allwinner D1.
-pub unsafe fn kernel_spim1(spi1: SPI_DBI, ccu: &mut CCU, gpio: &mut GPIO) -> Spim1 {
+pub unsafe fn kernel_spim1(spi1: SPI_DBI, ccu: &mut Ccu, gpio: &mut GPIO) -> Spim1 {
     // Set clock rate (fixed to 2MHz), and enable the SPI peripheral
-    ccu.spi1_clk.write(|w| {
+    ccu.borrow_raw().spi1_clk.write(|w| {
```

Comment:
Might want to add a TODO here to link to the "we shouldn't need `borrow_raw()` in the long term"